agreement/agreement.go

```diff
@@ -430,6 +424,13 @@ func (w *AgreementWorker) CommandHandler(command worker.Command) bool {
 			deleteMessage = false
 		} else if pDevice, err := persistence.FindExchangeDevice(w.db); err != nil {
 			glog.Errorf(logString(fmt.Sprintf("unable to get device from the local database. %v", err)))
+		} else if there, err := w.messageInExchange(exchangeMsg.MsgId); err != nil {
```

Comment:
I would rather see this new code before the FindExchangeDevice call.

Reply:
Changed per request